L1Trigger/L1THGCal/plugins/ntuples/HGCalTriggerNtupleHGCDigis.cc

```diff
@@ -219,6 +289,16 @@ simhits(const edm::Event& e, std::unordered_map<uint32_t, double>& simhits_ee, s
         auto itr_insert = simhits_fh.emplace(HGCalDetId(mysubdet,zp,layer,subsec,sec,cell), 0.);
         itr_insert.first->second += simhit.energy();
       }      
+      //  BH
+      int z=0, depth0=0, eta0=0, phi0=0, lay=0;
+      for( const auto& simhit : bh_simhits ) { 
+	HcalTestNumbering::unpackHcalIndex(simhit.id(), subdet, z, depth0, eta0, phi0, lay);
+	int sign = (z==0 ? -1 : 1);
+	HcalDDDRecConstants::HcalID tempid = triggerGeometry_->bhTopology().dddConstants()->getHCID(subdet, sign*eta0, phi0, lay, depth0);
```

Comment:
Rather than do all this by hand, you can just use the relabeller, e.g.:
https://github.com/cms-sw/cmssw/blob/master/SimCalorimetry/HGCalSimProducers/src/HGCDigitizer.cc#L58